### ISSUE TITLE
Add docs for user-defined kinds

### ISSUE THREAD

**Issue by chexxor** [purescript/documentation#39]
We recently [merged user-defined kinds](https://github.com/purescript/purescript/pull/2486), but I don't see any docs here for it. Is "foreign import kind" the new syntax? I would expect to find the docs in [language/FFI.md](https://github.com/purescript/documentation/blob/349a9e1de6e30defbe9afbc8652d941d2a754b94/language/FFI.md).

**Comment by hdgarrood** [comment]
Yes: https://github.com/purescript/purescript-typelevel-prelude/blob/v1.0.0/src/Type/Data/Ordering.purs#L18-L18

I'm not sure the FFI page is the right place for it, though; I don't think it really has anything to do with the FFI (in fact I think it's a bit unfortunate that the syntax suggests that it is). I think language/Syntax.md is a more appropriate place for it, and also the "Kind System" heading in language/Types.md will need some minor adjustments.
